pkg/policy/manager.go

```diff
@@ -102,13 +102,14 @@ func (m *ManagerImpl) Run(ctx context.Context, request *ExecuteRequest) (*Execut
 		}
 	}
 
-	var selector []string
-	if request.Policy.meta.subPolicy != "" {
-		selector = strings.Split(request.Policy.meta.subPolicy, "/")
+	filteredPolicy := request.Policy.Filter(request.Policy.meta.subPolicy)
+	if len(filteredPolicy.Policies.All()) == 0 && len(filteredPolicy.Checks) == 0 {
```

Comment:
nit: maybe make this a method? HasChecks()? which checks if policy has checks or any of its sub policies have checks

Reply:
I like this idea

Reply:
Implemented with tests and replaced all of the manual checks